pkg/alerter/alerter.go

```diff
@@ -0,0 +1,79 @@
+package alerter
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/sirupsen/logrus"
+)
+
+type Alerter struct {
+	sync.Mutex
+	Alerts   Alerts
+	AlertsCh chan Alerts
+}
+
+func NewAlerter() *Alerter {
+	return &Alerter{
+		Alerts:   make(Alerts),
+		AlertsCh: make(chan Alerts),
+	}
+}
+
+func (a *Alerter) Start() {
+	go func() {
+		defer close(a.AlertsCh)
```

Comment:
I think you should do something like this instead

```go
    for alert := range a.AlertsCh {
        for k, v := range alert {
		if val, ok := a.Alerts[k]; ok {
			a.Alerts[k] = append(val, v...)
		} else {
			a.Alerts[k] = v
		}
	}
    }
```

Since it is the only place we mutate the alert list we don't need a mutex.
When you'll close the AlertCh, the loop will break.

Reply:
Do you mean that I can get rid of the Mutex composition struct and to this instead ?

Reply:
I just tried it on indeed it works. I was afraid that with the goroutines it would have messed up the alerts map.

Reply:
The goroutine is only an async task which read from the chan and append to a list, this is thread safe unless you launch multiples instances of this routine to mutate the map.

This can be viewed as a single thead queue consumer inside your struct. So no concurrency issues there

Reply:
Done 7f5211c